provider/core/src/request.rs

```diff
@@ -264,56 +267,68 @@ impl DataLocale {
     /// use std::cmp::Ordering;
     ///
     /// let bcp47_strings: &[&str] = &[
+    ///     "ca",
+    ///     "ca$EUR",
     ///     "ca-ES",
-    ///     "ca-ES|USD", // TODO: Support parsing of this
+    ///     "ca-ES$GBP",
+    ///     "ca-ES$USD",
     ///     "ca-ES-u-ca-buddhist",
     ///     "ca-ES-valencia",
     ///     "cat",
-    ///     "ca|EUR",
     ///     "pl-Latn-PL",
     ///     "und",
+    ///     "und$MXN",
     ///     "und-fonipa",
     ///     "und-u-ca-hebrew",
     ///     "und-u-ca-japanese",
-    ///     "und|MXN",
     ///     "zh",
     /// ];
     ///
     /// for ab in bcp47_strings.windows(2) {
     ///     let a = ab[0];
     ///     let b = ab[1];
-    ///     assert!(a.cmp(b) == Ordering::Less);
-    ///     let a_loc: DataLocale = a.parse::<Locale>().unwrap().into();
-    ///     assert!(
-    ///         a_loc.strict_cmp(a.as_bytes()) == Ordering::Equal,
-    ///         "{} == {}",
-    ///         a,
+    ///     assert_eq!(a.cmp(b), Ordering::Less, "strings: {} < {}", a, b);
+    ///     let a_loc: DataLocale = a.parse().unwrap();
+    ///     assert_eq!(
+    ///         a_loc.strict_cmp(a.as_bytes()),
+    ///         Ordering::Equal,
+    ///         "strict_cmp: {} == {}",
+    ///         a_loc,
     ///         a
     ///     );
-    ///     assert!(
-    ///         a_loc.strict_cmp(b.as_bytes()) == Ordering::Less,
-    ///         "{} < {}",
-    ///         a,
+    ///     assert_eq!(
+    ///         a_loc.strict_cmp(b.as_bytes()),
+    ///         Ordering::Less,
+    ///         "strict_cmp: {} < {}",
+    ///         a_loc,
     ///         b
     ///     );
-    ///     let b_loc: DataLocale = b.parse::<Locale>().unwrap().into();
-    ///     assert!(
-    ///         b_loc.strict_cmp(b.as_bytes()) == Ordering::Equal,
-    ///         "{} == {}",
-    ///         b,
+    ///     let b_loc: DataLocale = b.parse().unwrap();
+    ///     assert_eq!(
+    ///         b_loc.strict_cmp(b.as_bytes()),
+    ///         Ordering::Equal,
+    ///         "strict_cmp: {} == {}",
+    ///         b_loc,
     ///         b
     ///     );
-    ///     assert!(
-    ///         b_loc.strict_cmp(a.as_bytes()) == Ordering::Greater,
-    ///         "{} > {}",
-    ///         b,
+    ///     assert_eq!(
+    ///         b_loc.strict_cmp(a.as_bytes()),
+    ///         Ordering::Greater,
+    ///         "strict_cmp: {} > {}",
+    ///         b_loc,
     ///         a
     ///     );
     /// }
     /// ```
     pub fn strict_cmp(&self, other: &[u8]) -> Ordering {
-        // TODO: Make this work with aux keys
-        let subtags = other.split(|b| *b == b'-');
+        let mut pipe_iter = other.split(|b| *b == b'$');
```

Comment:
nit: let's put the aux key separator in a constant somewhere

Reply:
Fixed by adding `pub const fn AuxiliaryKey::separator()`